native/core/src/execution/datafusion/expressions/comet_scalar_funcs.rs

```diff
@@ -0,0 +1,186 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This file is an extract of what used to be `scalar_funcs.rs`. The `create_comet_physical_expr` isn't easily reusable for others so it seems reasonable to keep it here.

Reply:
This looks good. I wonder if having `comet` in the file name is a bit redundant though

Reply:
It kinda is, but I had it there just to distinguish from `spark-expr/src/scalar_funcs.rs` and as what this file does is related to CometScalarUDFs 🤷